src/components/searchUserAside/index.js

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from "react"
+import { NavLink } from "react-router-dom"
+
+import Button from "../button"
+import Card from "../card"
+import Menu from "../menu"
+import UsersList from "../usersList"
+import UserSearchBar from "../userSearchBar/UserSearchBar"
+
+import { getUserByName } from "../../service/apiClient"
+
+import "./style.css"
+
+const SearchUserAside = () => {
+  const [hasFocus, setHasFocus] = useState(false)
+  const [menuHover, setMenuHover] = useState(false)
+  const [searchVal, setSearchVal] = useState("")
+  const [users, setUsers] = useState([])
+
+  const menuOpen = hasFocus || searchVal.length > 0 || menuHover
+
+  const getUsers = () => {
+    getUserByName(searchVal).then(setUsers)
+  }
+
+  useEffect(getUsers, [searchVal])
+
+  return (
+    <div className="relative">
+      <Card>
+        <form
+          onFocus={() => setHasFocus(true)}
+          onBlur={() => setHasFocus(false)}
+        >
+          <UserSearchBar searchVal={searchVal} setSearchVal={setSearchVal} />
+        </form>
+      </Card>
+      {menuOpen && (
+        <UserResults
+          users={users}
+          menuHover={menuHover}
+          setMenuHover={setMenuHover}
+        />
+      )}
+    </div>
+  )
+}
+
+const UserResults = ({ users, menuHover, setMenuHover }) => {
+  return (
+    <>
+      <div
+        onMouseEnter={() => setMenuHover(true)}
+        onMouseLeave={() => setMenuHover(false)}
+      >
+        <Menu className={"search-user-menu"}>
+          <p className="border-bottom spacing text-blue1">People</p>
+          <UsersList users={users.slice(0, 10)} />
```

Comment:
might want to check if users is defined and is a string before this return block to prevent an error